fedn/fedn/network/controller/control.py

```diff
@@ -220,12 +244,10 @@ def round(self, session_config, round_id):
         else:
             print("REDUCER: failed to update model in round with config {}".format(
                 session_config), flush=True)
-            round_data['status'] = 'Failed'
-            return None, round_data
+            self.set_round_status(round_id, 'Failed')
+            return None, self.statestore.get_round(round_id)
 
-        round_data['status'] = 'Success'
-
-        # 4. Trigger participating combiner nodes to execute a validation round for the current model
+        # Ask combiners to validate the new global model
         validate = session_config['validate']
         if validate:
```

Comment:
I think we should decouple this from from "round" function. Maybe have "round_training" and "round_validate"?

Reply:
I agree, but this should be handled in a separate PR. Let's put on the backlog.